src/services/CicService.ts

```diff
@@ -306,29 +306,30 @@ export class CicService {
 		}
 	}
 
-	private mapCICNames(givenNames: string[], familyNames: string[]): PersonIdentityName[] {
+	private mapCICNames(givenNames: string[], familyName: string): PersonIdentityName[] {
 		const nameParts: PersonIdentityNamePart[] = [];
+	
+		const validateName = (name: string) => {
+			if (!Constants.GIVEN_NAME_REGEX.exec(name)) {
+				throw new AppError(`Given name doesn't match regex expression: ${Constants.GIVEN_NAME_REGEX}`, HttpCodesEnum.BAD_REQUEST);
```

Comment:
Logging an error is needed along with a new `messageCode` please.
Also, `name` passed in could be a Given name or Family name so worth keeping it generic

Reply:
`messageCode` has been added and error message changed to "Name doesn't match..." to be more generic